File: .skills/openclaw-skills/skills/harrylabs0913/health-manager/src/database/connection.ts
import Database from 'better-sqlite3';
import path from 'path';
import os from 'os';
import fs from 'fs';
import { SCHEMA, DEFAULT_CONFIG } from './schema';
import { initDefaultReminders } from './reminders';

const DATA_DIR = path.join(os.homedir(), '.config', 'health-manager');
const DB_PATH = path.join(DATA_DIR, 'health.db');

let db: Database.Database | null = null;

/**
 * 初始化数据目录
 */
function initDataDir(): void {
  if (!fs.existsSync(DATA_DIR)) {
    fs.mkdirSync(DATA_DIR, { recursive: true });
  }
}

/**
 * 获取数据库连接
 */
export function getDatabase(): Database.Database {
  if (!db) {
    initDataDir();
    db = new Database(DB_PATH);
    db.pragma('journal_mode = WAL');
    initializeSchema();
  }
  return db;
}

/**
 * 初始化数据库表结构
 */
function initializeSchema(): void {
  if (!db) return;

  // 创建表
  Object.values(SCHEMA).forEach(sql => {
    db!.exec(sql);
  });

  // 插入默认配置
  const insertConfig = db.prepare(`
    INSERT OR IGNORE INTO config (key, value, description)
    VALUES (@key, @value, @description)
  `);

  DEFAULT_CONFIG.forEach(config => {
    insertConfig.run(config);
  });

  // 初始化默认提醒
  initDefaultReminders();
}

/**
 * 关闭数据库连接
 */
export function closeDatabase(): void {
  if (db) {
    db.close();
    db = null;
  }
}

/**
 * 获取数据库文件路径
 */
export function getDatabasePath(): string {
  return DB_PATH;
}

export { initDefaultReminders };

File: .skills/openclaw-skills/skills/harrylabs0913/reading-manager/src/reading_database/connection.py
"""
数据库连接管理
"""
import sqlite3
import os
from pathlib import Path
from .schema import SCHEMAS, DEFAULT_CONFIG

DATA_DIR = Path.home() / ".config" / "reading-manager"
DB_PATH = DATA_DIR / "reading.db"


def init_data_dir():
    """初始化数据目录"""
    DATA_DIR.mkdir(parents=True, exist_ok=True)


def get_connection():
    """获取数据库连接"""
    init_data_dir()
    conn = sqlite3.connect(DB_PATH)
    conn.row_factory = sqlite3.Row
    return conn


def init_database():
    """初始化数据库"""
    conn = get_connection()
    cursor = conn.cursor()
    
    # 创建表和索引
    for schema in SCHEMAS:
        # 分割多个 SQL 语句
        statements = [s.strip() for s in schema.split(';') if s.strip()]
        for stmt in statements:
            cursor.execute(stmt)
    
    # 插入默认配置
    for key, value, description in DEFAULT_CONFIG:
        cursor.execute(
            "INSERT OR IGNORE INTO config (key, value, description) VALUES (?, ?, ?)",
            (key, value, description)
        )
    
    conn.commit()
    conn.close()


def get_db_path():
    """获取数据库路径"""
    return str(DB_PATH)
